kafka_consumer/datadog_checks/kafka_consumer/client.py

```diff
@@ -0,0 +1,245 @@
+        # https://github.com/confluentinc/confluent-kafka-python/issues/594
+        self.kafka_client.list_topics(None, timeout=self.config._request_timeout_ms / 1000)
+
+    def get_consumer_offsets(self):
+        # {(consumer_group, topic, partition): offset}
+        consumer_offsets = {}
+
+        if self.config._monitor_unlisted_consumer_groups:
+            # Get all consumer groups
+            consumer_groups = []
+            consumer_groups_future = self.kafka_client.list_consumer_groups()
+            self.log.debug('MONITOR UNLISTED CG FUTURES: %s', consumer_groups_future)
+            try:
+                list_consumer_groups_result = consumer_groups_future.result()
+                self.log.debug('MONITOR UNLISTED FUTURES RESULT: %s', list_consumer_groups_result)
+
+                consumer_groups.extend(
+                    valid_consumer_group.group_id for valid_consumer_group in list_consumer_groups_result.valid
+                )
+            except Exception as e:
+                self.log.error("Failed to collect consumer offsets %s", e)
+
+        elif self.config._consumer_groups:
+            self._validate_consumer_groups()
+            consumer_groups = self.config._consumer_groups
+
+        else:
+            raise ConfigurationError(
+                "Cannot fetch consumer offsets because no consumer_groups are specified and "
+                "monitor_unlisted_consumer_groups is %s." % self.config._monitor_unlisted_consumer_groups
+            )
+
+        for future in self._get_consumer_offset_futures(consumer_groups):
+            try:
+                response_offset_info = future.result()
+                self.log.debug('FUTURE RESULT: %s', response_offset_info)
+                consumer_group = response_offset_info.group_id
+                topic_partitions = response_offset_info.topic_partitions
+
+                self.log.debug('RESULT CONSUMER GROUP: %s', consumer_group)
+                self.log.debug('RESULT TOPIC PARTITIONS: %s', topic_partitions)
+
+                for topic_partition in topic_partitions:
+                    topic = topic_partition.topic
+                    partition = topic_partition.partition
+                    offset = topic_partition.offset
+
+                    self.log.debug('RESULTS TOPIC: %s', topic)
+                    self.log.debug('RESULTS PARTITION: %s', partition)
+                    self.log.debug('RESULTS OFFSET: %s', offset)
+
+                    if topic_partition.error:
+                        self.log.debug(
+                            "Encountered error: %s. Occurred with topic: %s; partition: [%s]",
+                            topic_partition.error.str(),
+                            topic_partition.topic,
+                            str(topic_partition.partition),
+                        )
+                    consumer_offsets[(consumer_group, topic, partition)] = offset
+            except KafkaException as e:
```

Comment:
Same thing about trying to keep try-except blocks small, this one looks like could be moved way up.